src/Compilers/CSharp/Portable/Lowering/LocalRewriter/LocalRewriter_NullCoalescingAssignmentOperator.cs

```diff
@@ -15,20 +15,17 @@ public override BoundNode VisitNullCoalescingAssignmentOperator(BoundNullCoalesc
 
             // Rewrite LHS with temporaries to prevent double-evaluation of side effects, as we'll need to use it multiple times.
             BoundExpression transformedLHS = TransformCompoundAssignmentLHS(node.LeftOperand, stores, temps, node.LeftOperand.HasDynamicType());
-            var lhsRead = TransformNullCoalescingLHSRead(transformedLHS, stores, temps);
+            var lhsRead = MakeRValue(transformedLHS);
             BoundExpression loweredRight = VisitExpression(node.RightOperand);
 
-            // Now that LHS is transformed with temporaries, we rewrite this node into a conditional expression:
-            // (lhsRead != null) ? lhsRead : (transformedLHS = rhs)
-
-            // lhsRead != null
-            BoundExpression nullCheck = MakeNullCheck(syntax, lhsRead, BinaryOperatorKind.NotEqual);
+            // Now that LHS is transformed with temporaries, we rewrite this node into a coalesce expression:
+            // lhsRead ?? (transformedLHS = loweredRight)
 
             // transformedLHS = rhs
             BoundExpression assignment = MakeAssignmentOperator(syntax, transformedLHS, loweredRight, node.LeftOperand.Type, used: true, node.IsChecked, isCompoundAssignment: true);
 
-            // (lhsRead != null) ? lhsRead : (transformedLHS = rhs)
-            BoundExpression conditionalExpression = RewriteConditionalOperator(syntax, nullCheck, lhsRead, assignment, constantValueOpt: null, rewrittenType: node.LeftOperand.Type, isRef: false);
+            // lhsRead ?? (transformedLHS = loweredRight)
+            BoundExpression conditionalExpression = MakeNullCoalescingOperator(syntax, lhsRead, assignment, Conversion.Identity, node.LeftOperand.Type);
```

Comment:
Are tuple conversions identity conversions? It may be good to test something like

```C#
{
  (int x, int y)? t1 = null;
  t1 ??= (a: 0, b: 0);
}
```

Reply:
Added a item to my tracking list of tests to add. I'll do that in a follow-up PR.

---
In reply to: [203174229](https://github.com/dotnet/roslyn/pull/28598#discussion_r203174229) [](ancestors = 203174229)